app/components/UI/Swaps/useStablecoinsDefaultSlippage.test.tsx

```diff
@@ -0,0 +1,190 @@
+          destTokenAddress: '0xc2132d05d31c914a87c6611c10748aeb04b58e8f', // USDT
+          chainId: swapsUtils.POLYGON_CHAIN_ID as Hex,
+          setSlippage: mockSetSlippage,
+        }),
+      { state: initialState },
+    );
+
+    expect(mockSetSlippage).toHaveBeenCalledWith(0.5);
+  });
+
+  it('should not set slippage when source token is not on the list of stablecoins', () => {
+    renderHookWithProvider(
+      () =>
+        useStablecoinsDefaultSlippage({
+          sourceTokenAddress: '0x6b175474e89094c44da98b954eedeac495271d0f', // DAI (not in the list)
+          destTokenAddress: '0xdac17f958d2ee523a2206206994597c13d831ec7', // USDT
+          chainId: swapsUtils.ETH_CHAIN_ID as Hex,
+          setSlippage: mockSetSlippage,
+        }),
+      { state: initialState },
+    );
+
+    expect(mockSetSlippage).not.toHaveBeenCalled();
+  });
+
+  it('should not set slippage when destination token is not on the list of stablecoins', () => {
+    renderHookWithProvider(
+      () =>
+        useStablecoinsDefaultSlippage({
+          sourceTokenAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48', // USDC
+          destTokenAddress: '0x6b175474e89094c44da98b954eedeac495271d0f', // DAI (not in the list)
+          chainId: swapsUtils.ETH_CHAIN_ID as Hex,
+          setSlippage: mockSetSlippage,
+        }),
+      { state: initialState },
+    );
+
+    expect(mockSetSlippage).not.toHaveBeenCalled();
+  });
+
+  it('should not set slippage when chain ID is not supported', () => {
+    renderHookWithProvider(
+      () =>
+        useStablecoinsDefaultSlippage({
+          sourceTokenAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48', // USDC
+          destTokenAddress: '0xdac17f958d2ee523a2206206994597c13d831ec7', // USDT
+          chainId: '0x9999' as Hex, // Unsupported chain ID
+          setSlippage: mockSetSlippage,
+        }),
+      { state: initialState },
+    );
+
+    expect(mockSetSlippage).not.toHaveBeenCalled();
+  });
+
+  it('should not set slippage when source token address is missing', () => {
+    renderHookWithProvider(
+      () =>
+        useStablecoinsDefaultSlippage({
+          destTokenAddress: '0xdac17f958d2ee523a2206206994597c13d831ec7', // USDT
+          chainId: swapsUtils.ETH_CHAIN_ID as Hex,
+          setSlippage: mockSetSlippage,
+        }),
+      { state: initialState },
+    );
+
+    expect(mockSetSlippage).not.toHaveBeenCalled();
+  });
+
+  it('should not set slippage when destination token address is missing', () => {
+    renderHookWithProvider(
+      () =>
+        useStablecoinsDefaultSlippage({
+          sourceTokenAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48', // USDC
+          chainId: swapsUtils.ETH_CHAIN_ID as Hex,
+          setSlippage: mockSetSlippage,
+        }),
+      { state: initialState },
+    );
+
+    expect(mockSetSlippage).not.toHaveBeenCalled();
+  });
+
+  it('should call setSlippage only once when dependencies change', () => {
```

Comment:
Nit: Test description renaming
```suggestion
  it('calls setSlippage only once when dependencies change', () => {
```

Similar for other tests